hack/crossplane/azure-crossplane.yaml

```diff
@@ -11,37 +12,77 @@ typeFilters:
     name: Expression
     because: expression is an ARM template construct which doesn't belong in CRDs
   - action: include
-    group: microsoft.common
+    group: common
     matchRequired: false # only triggers for Status types
   - action: include
     version: v*api20201101
-    group: microsoft.network
-    because: Including microsoft.network for Crossplane
+    group: network
+    because: Including network for Crossplane
   - action: include
     version: v*api20200601
-    group: microsoft.cache
-    because: Including microsoft.cache for Crossplane
+    group: cache
+    because: Including cache for Crossplane
   - action: include
     version: v*api20201101preview
-    group: microsoft.sql
-    because: Including microsoft.sql
+    group: sql
+    because: Including sql
   - action: include
-    group: microsoft.sql
+    group: sql
     version: '*v1'
-    because: Including microsoft.sql v1 common types
+    because: Including sql v1 common types
     matchRequired: false # only triggers for Status types
   - action: include
-    group: microsoft.sql
+    group: sql
     version: '*v2'
-    because: Including microsoft.sql v2 common types
+    because: Including sql v2 common types
     matchRequired: false # only triggers for Status types
   - action: prune
     version: '*preview'
     because: preview SDK versions are excluded by default
   - action: prune
-    group: 'microsoft.*'
+    group: network
     version: '*'
-    because: Prune everything else
+    because: Pruning all networking api versions since we don't use them anyway
```

Comment:
Before, we were pruning everything that had a group of `microsoft.*`. When I tried to prune `*` instead, I found that it was pruning more than before. I believe that this is because there are some things in groups like `common` that never had a microsoft prefix (and are used solely in `allOf`/`oneOf` type situations). Pruning those things breaks the generator, so I moved it to have a configuration more like `azure-arm` that prunes the problem things but leaves "most" of the rest.